### PR TITLE
enable/disable chat sounds partially - console command only

### PR DESCRIPTION
**Pull request by jxsl13**
Please squash merge this, as there is crap in my commit history.

Enable chat sound for:
0=no client messages create a sound effect
1=whisper only creates a sound effect
2=if your nickname is tagged, a sound effect is created 
3=whisper or nickname tagged
4=whisper, nickname tagged & server msg 
5=whisper, nickname tagged, server msg  & clientside messages(adding players as friends via /friend <..>) 
6=all chat messages create a chat sound

### CODE REVIEW

**Comment by ChillerDragon** [comment]
Where is the option to use 0.6.3 sounds :c

**Comment by heinrich5991** [comment]
> Where is the option to use 0.6.3 sounds :c

This is unrelated to this pull request.

File: src/game/variables.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ MACRO_CONFIG_INT(ClAutoswitchWeapons, cl_autoswitch_weapons, 0, 0, 1, CFGFLAG_CL
 MACRO_CONFIG_INT(ClShowhud, cl_showhud, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "Show ingame HUD")
 MACRO_CONFIG_INT(ClFilterchat, cl_filterchat, 0, 0, 2, CFGFLAG_CLIENT|CFGFLAG_SAVE, "Show chat messages from: 0=all, 1=friends only, 2=no one")
 MACRO_CONFIG_INT(ClShowsocial, cl_showsocial, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "Show social data like names, clans, chat etc.")
+MACRO_CONFIG_INT(ClChatSound, cl_chatsound, 6, 0, 6, CFGFLAG_CLIENT|CFGFLAG_SAVE, "Enable chat sound for 0=no msgs 1=whisper only 2=tagged 3=(1) & (2) 4=(3) & server msg 5=(4) & clientside msgs 6=all chat msgs")
```
Comment (Dune-jr) [inline src/game/variables.h]: For the lambda user, I think this is very confusing. Maybe it would be better to split those? I don't know.

**Review by Dune-jr — COMMENTED** [review]
Looks just fine, I'm only a bit skeptical of this level of customisation in the default client (especially since you can already achieve a lot by removing audio files that you dislike).

File: src/game/variables.h (hunk 1 above)
Comment (jxsl13) [reply src/game/variables.h]: 128 visible characters, had not much spare space there, I guess.

**Comment by jxsl13** [comment]
As it's only a console command I think that whoever wants to customize their client beyond the non-cluttered user interface, can search for the appropriate console commands to do so.

**Comment by oy** [comment]
Using flags would probably be better: whisper, highlight, server msg, client msg. So you can customize as you like.

> Where is the option to use 0.6.3 sounds :c

There's no difference.

**Comment by ChillerDragon** [comment]
@oy the souns were swapped in 0.6.4 the highlight and normal chat sound. And most players are really confused by this. So a config to revert this would be nice. But as heinrich already said its probably unrelated.

**Comment by oy** [comment]
@ChillerDragon no 0.6.3 already had it that way. Change was with 0.6.2 6649d651a8fcb170ccfd53c30d1f3c55d86b9767 6 years ago.

**Comment by jxsl13** [comment]
Suggestion:
```C++
MACRO_CONFIG_INT(m_ClChatsoundWhisper, cl_chatsound_whisper, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "You will hear a sound, if you receive a whisper message.")
MACRO_CONFIG_INT(ClChatsoundHighlight, cl_chatsound_highlight, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "You will hear a sound, if your nickname is tagged by another player.")
MACRO_CONFIG_INT(ClChatsoundServerMessage, cl_chatsound_server_msg, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "You will hear a sound if the server sends you a message e.g. when someone joins the server.")
MACRO_CONFIG_INT(ClChatsoundClientMessage, cl_chatsound_client_msg, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "You will hear a sound if someone writes in the chat.")
MACRO_CONFIG_INT(ClChatsoundLocalMessage, cl_chatsound_local_msg, 1, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "You will hear a sound if you e.g. add a player as friend.")
```

**Comment by oy** [comment]
@jxsl13 put that all in one config command 1.bit=whisper, 2.bit=highlight, 3.bit=server msg, 4.bit=client msg. What's local message?

**Comment by jxsl13** [comment]
does it make sense for a normal user to do bitwise arithmetic to get their stuff configured?
I changed that slightly to client message for a better differentiation.

**Comment by oy** [comment]
It's done with other configs too, just 2 or 3 bit probably. But people should be able to handle it. Could put some more info's in the docs.

**Comment by Dune-jr** [comment]
It would be nice if we had some sort of UI mechanism that lets us configure advanced options in a popup or something. I had the issue with the statboard, used this slightly unsatisfactory solution:

image

**Comment by jxsl13** [comment]
Where is it done with flags, wanna see for myself as a reference :O?

**Comment by jxsl13** [comment]
@Dune-jr @oy 
I'd do it like this:
https://github.com/jxsl13/zcatch/commit/bf27068f8d77b144314e0e5fd1945292ec731223

**Comment by jxsl13** [comment]
Updated to the latest variant: https://github.com/jxsl13/zcatch/commit/42034e454296bb0ef2770de2b275944a5645dc3f
please review.

**Comment by Fudgyking** [comment]
As Dune already pointed out, I think those settings look way too precise to be put in the vanilla version. It just looks like something maybe 2 people would use.

**Comment by LordSk** [comment]
So what is the use case here? How often do you enable/disable each chat sound?
* Do you disable a select few sounds permanently?
* Do you switch between 2 modes, let's say "silent chat" and "full chat"? More modes perhaps?
* Do you change your selection of chat sounds every game, depending on the mood?

The solution you're proposing here covers all of those, sure. But do we want to cover all of those?
It also add quite a bit of complexity to the code (is it worth it?) *and* to the potential user.
Most people will never touch this setting, and I would argue we should make it so they don't even want to think about disabling/enabling chat sounds.

I hate to say it, but I don't think this PR is suited for Teeworlds right now. However if people really want the ability to customize chat sounds, we should definitely revisit the topic and perhaps try a few other directions (adjust default individual volume sounds? tweak audio files? etc...).

**Comment by jxsl13** [comment]
Some kinda tw.com poll maybe?

**Comment by Dune-jr** [comment]
I echo @LordSk's sentiment in that if anything, there should be a clearly identified issue — for example, server message sounds are annoying in some situation — and a well-designed fix (better spam protection, better audio, mute server, or disable sounds like you suggest). It is dangerously tempting to make a patchwork Teeworlds *à la carte* where the smallest thing is an option. 

https://github.com/jxsl13/zcatch/commit/42034e454296bb0ef2770de2b275944a5645dc3f is +110 -26, includes regex, string, sstream. That is not benign. 
Maybe this is kind of customization code that is be best left to modifications and merged in when there is popular demand.

**Comment by jxsl13** [comment]
latest commit removes regex, string & sstream.

**Comment by oy** [comment]
Don't think that a UI setting should be added for it cause there's probably just a small group of users who would use it. 
As mentioned by others, dunno if there's a public demand for such a feature at all. Would be good to find that out.

**Comment by fokkonaut** [comment]
Can just delete the wanted sound files, just as you can swap the sound from msg chat and highlight again.

**Pull request closed without merging**